src/help.cpp

```diff
@@ -157,136 +115,216 @@ std::string help::get_note_colors()
-                                            1 + static_cast<int>( TERMY > FULL_SCREEN_HEIGHT ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0 ) ) );
-        ui.position_from_window( w_help_border );
-    };
-    init_windows( ui );
-    ui.on_screen_resize( init_windows );
-
-    input_context ctxt( "DISPLAY_HELP", keyboard_mode::keychar );
-    ctxt.register_cardinal();
+    ctxt = input_context( "DISPLAY_HELP", keyboard_mode::keychar );
     ctxt.register_action( "QUIT" );
     ctxt.register_action( "CONFIRM" );
-    // for mouse selection
+    // Mouse selection
     ctxt.register_action( "SELECT" );
     ctxt.register_action( "MOUSE_MOVE" );
-    // for the menu shortcuts
+    // Generated shortcuts
     ctxt.register_action( "ANY_INPUT" );
-
-    std::string action;
-    std::map<int, inclusive_rectangle<point>> opt_map;
-    int sel = -1;
-
+    // Switching between categories while open
+    ctxt.register_leftright();
+    ctxt.register_action( "PREV_TAB" );
+    ctxt.register_action( "NEXT_TAB" );
     const hotkey_queue &hkq = hotkey_queue::alphabets();
     input_event next_hotkey = ctxt.first_unassigned_hotkey( hkq );
-    std::map<int, input_event> hotkeys;
-    for( const auto &text : help_texts ) {
+    for( const auto &text : data.help_categories ) {
         hotkeys.emplace( text.first, next_hotkey );
         next_hotkey = ctxt.next_unassigned_hotkey( hkq, next_hotkey );
     }
+}
+
+void help_window::draw_controls()
+{
+    if( !has_selected_category ) {
+        draw_category_selection();
+    } else {
+        draw_category();
+    }
+}
 
-    ui.on_redraw( [&]( const ui_adaptor & ) {
-        draw_border( w_help_border, BORDER_COLOR, _( "Help" ) );
-        wnoutrefresh( w_help_border );
-        opt_map = draw_menu( w_help, sel, hotkeys );
-    } );
-
-    do {
-        ui_manager::redraw();
-
-        sel = -1;
-        action = ctxt.handle_input();
-        input_event input = ctxt.get_raw_input();
-
-        // Mouse selection
-        if( action == "MOUSE_MOVE" || action == "SELECT" ) {
-            std::optional<point> coord = ctxt.get_coordinates_text( w_help );
-            if( !!coord ) {
-                int cnt = run_for_point_in<int, point>( opt_map, *coord,
-                [&sel]( const std::pair<int, inclusive_rectangle<point>> &p ) {
-                    sel = p.first;
-                } );
-                if( cnt > 0 && action == "SELECT" ) {
-                    auto iter = hotkeys.find( sel );
-                    if( iter != hotkeys.end() ) {
-                        input = iter->second;
-                        action = "CONFIRM";
-                    }
-                }
+void help_window::draw_category_selection()
+{
+    // TODO: Add one column display for tiny screens and screen reader users
+    selected_option = -1;
+    //~ Help menu header
+    format_title( _( "Help" ) );
+    // Split the categories in half
+    if( ImGui::BeginTable( "Category Options", 3, ImGuiTableFlags_None ) ) {
```

Comment:
```suggestion
    if( ImGui::BeginTable( "Category Options", 2, ImGuiTableFlags_None ) ) {
```

```suggestion
    if( ImGui::BeginTable( "Category Options", 2, ImGuiTableFlags_None ) ) {
```

That's why you had that mystery third column and why calling `ImGui::TableNextRow` fixed it by simply skipping the third column every time.